### PR TITLE
Remove code used for stashing - add supervision classes to handle existing stashed elements deletion

### PR DESCRIPTION
**Pull request by klesaulnier**

### CODE REVIEW

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/directory/server/services/DirectoryRepositoryService.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public List<DirectoryElementEntity> getElementEntities(List<UUID> elementUuids)
         return directoryElementRepository.findAllByIdIn(elementUuids);
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/main/java/org/gridsuite/directory/server/services/DirectoryRepositoryService.java]: fixed

File: src/test/java/org/gridsuite/directory/server/utils/DirectoryTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.gridsuite.directory.server.utils;
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/test/java/org/gridsuite/directory/server/utils/DirectoryTestUtils.java:7]: fixed

File: src/test/java/org/gridsuite/directory/server/DirectoryElementRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package org.gridsuite.directory.server;
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/test/java/org/gridsuite/directory/server/DirectoryElementRepositoryTest.java:7]: fixed

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 4]
```diff
@@ -434,6 +424,46 @@ private void deleteSubElements(UUID elementUuid, String userId) {
...
+            .map(ElementAttributes::toElementAttributes)
+            .filter(element -> !DIRECTORY.equals(element.getType()))
+            .toList();
+
+        // deleting all elements
+        repositoryService.deleteElements(elementsAttributesToDelete.stream().map(ElementAttributes::getElementUuid).toList());
+
+        // extracting studyUuids from this list, to send specific notifications
+        List<UUID> studyToDeleteUuids = elementsAttributesToDelete.stream().filter(element -> STUDY.equals(element.getType())).map(ElementAttributes::getElementUuid).toList();
+        studyToDeleteUuids.forEach(studyUuid -> notificationService.emitDeletedStudy(studyUuid, userId));
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply src/main/java/org/gridsuite/directory/server/DirectoryService.java]: fixed

**Pull request merged**